File: Server/src/main/content/global/skill/production/cooking/recipe/pizza/impl/MeatPizza.java
package content.global.skill.production.cooking.recipe.pizza.impl;

import content.global.skill.production.cooking.recipe.pizza.PizzaRecipe;
import core.game.node.item.Item;

/**
 * The Meat pizza.
 */
public class MeatPizza extends PizzaRecipe {

    private static final Item MEAT_PIZZA = new Item(2293);

    private static final Item COOKED_MEAT = new Item(2142);

    private static final Item COOKED_CHICKEN = new Item(2140);

    @Override
    public double getExperience() {
        return 26;
    }

    @Override
    public int getLevel() {
        return 45;
    }

    @Override
    public Item getProduct() {
        return MEAT_PIZZA;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{COOKED_MEAT, COOKED_CHICKEN};
    }

}

File: Emulator/src/main/content/global/skill/cooking/type/pizza/impl/MeatPizza.kt
package content.global.skill.cooking.type.pizza.impl

import content.global.skill.cooking.type.pizza.PizzaRecipe
import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Meat pizza.
 */
class MeatPizza : PizzaRecipe() {

    override fun getExperience(): Double {
        return 26.0
    }

    override fun getLevel(): Int {
        return 45
    }

    override fun getProduct(): Item {
        return MEAT_PIZZA
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(COOKED_MEAT, COOKED_CHICKEN)
    }

    companion object {
        private val MEAT_PIZZA = Item(Items.MEAT_PIZZA_2293)
        private val COOKED_MEAT = Item(Items.COOKED_MEAT_2142)
        private val COOKED_CHICKEN = Item(Items.COOKED_CHICKEN_2140)
    }
}
